src/lua/CommunityBalanceMod/Combined/DamageTypes.lua

```diff
@@ -60,48 +60,84 @@ function GetReceivesBiologicalDamage(entity)
+           -- [kTechId.Flamethrower]    = { kFlamethrowerWeapons1DamageScalar,    kFlamethrowerWeapons2DamageScalar,    kFlamethrowerWeapons3DamageScalar },
             ["Default"]               = { kWeapons1DamageScalar,                kWeapons2DamageScalar,                kWeapons3DamageScalar },
         }
 
     end
 
+    --MDS Marines
+    if not upgradedDamageScalarsStructure then
+
+        upgradedDamageScalarsStructure =
+        {
+            [kTechId.Shotgun]         = { kShotgunWeapons1DamageScalarStructure,         kShotgunWeapons2DamageScalarStructure,         kShotgunWeapons3DamageScalarStructure },
+           -- [kTechId.GrenadeLauncher] = { kGrenadeLauncherWeapons1DamageScalarStructure, kGrenadeLauncherWeapons2DamageScalarStructure, kGrenadeLauncherWeapons3DamageScalarStructure },
+           -- [kTechId.Flamethrower]    = { kFlamethrowerWeapons1DamageScalarStructure,    kFlamethrowerWeapons2DamageScalarStructure,    kFlamethrowerWeapons3DamageScalarStructure },
+            ["Default"]               = { kWeapons1DamageScalarStructure,                kWeapons2DamageScalarStructure,                kWeapons3DamageScalarStructure },
+        }
+
+    end
+
     local upgradeScalars = upgradedDamageScalars["Default"]
     if upgradedDamageScalars[weaponTechId] then
         upgradeScalars = upgradedDamageScalars[weaponTechId]
     end
 
-    if GetHasTech(attacker, kTechId.Weapons3, true) then
-        return upgradeScalars[3]
-    elseif GetHasTech(attacker, kTechId.Weapons2, true) then
-        return upgradeScalars[2]
-    elseif GetHasTech(attacker, kTechId.Weapons1, true) then
-        return upgradeScalars[1]
+    --MDS Marines
+    local upgradeScalarsStructure = upgradedDamageScalarsStructure["Default"]
+    if upgradedDamageScalarsStructure[weaponTechId] then
+        upgradeScalarsStructure = upgradedDamageScalarsStructure[weaponTechId]
     end
 
+    --MDS Marines
+    if structureDamage then 
+        if GetHasTech(attacker, kTechId.Weapons3, true) then
+            return upgradeScalarsStructure[3]
+        elseif GetHasTech(attacker, kTechId.Weapons2, true) then
+            return upgradeScalarsStructure[2]
+        elseif GetHasTech(attacker, kTechId.Weapons1, true) then
+            return upgradeScalarsStructure[1]
+        end
+    else
+
+
+        if GetHasTech(attacker, kTechId.Weapons3, true) then
+            return upgradeScalars[3]
+        elseif GetHasTech(attacker, kTechId.Weapons2, true) then
+            return upgradeScalars[2]
+        elseif GetHasTech(attacker, kTechId.Weapons1, true) then
+            return upgradeScalars[1]
+        end
+    end
+    
     return 1.0
 
 end
 
+-- MDS Marines
 -- Use this function to change damage according to current upgrades
-function NS2Gamerules_GetUpgradedDamage(attacker, doer, damage)
+function NS2Gamerules_GetUpgradedDamage(target, attacker, doer, damage)
```

Comment:
Same again but for `target`

Reply:
It only gets called in DamageType.lua
This should not cause any issues, but to be make sure its not getting called wrong by other mods we will append target to the last unused position of the function.

Reply:
Sounds good 👍